### PR TITLE
SAAS/3257: Added wisdomDomainEncrypted Plugin and spec

### PR DESCRIPTION
**Pull request by m-nouman-suleman**
## Task
https://scalock.atlassian.net/browse/SAAS-3257

### CODE REVIEW

File: plugins/aws/connect/wisdomDomainEncrypted.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    category: 'Connect',
+    domain: 'Content Delivery',
+    description: 'Ensure that Wisdom domains created under Amazon Connect instances are using desired KMS encryption level.',
+    more_info: 'All user data stored in Amazon Connect Wisdom is encrypted at rest using encryption keys stored in AWS Key Management Service. Additionaly, you can provide customer managed KMS keys in order to gain more control over encryption/decryption processes.',
+    recommended_action: 'Ensure that Amazon Connect Wisdom domains have encryption enabled.',
+    link: 'https://docs.aws.amazon.com/connect/latest/adminguide/encryption-at-rest.html',
+    apis: ['Wisdom:listAssistants', 'KMS:listKeys', 'KMS:describeKey'],
+    settings: {
+        wisdom_desired_encryption_level: {
+            name: 'Connect Wisdom Domain Encrypted',
```
Comment (AkhtarAmir) [inline plugins/aws/connect/wisdomDomainEncrypted.js:15]:
```suggestion
            name: 'Connect Wisdom Domain Desired Encryption Level',
```

File: plugins/aws/connect/wisdomDomainEncrypted.spec.js [hunk 2]
```diff
@@ -0,0 +1,176 @@
+var expect = require('chai').expect;
+var wisdomDomainEncrypted = require('./wisdomDomainEncrypted');
+
+const listAssistants = [
+    {
+      assistantArn: 'arn:aws:wisdom:us-east-1:101363889637:assistant/6bebcb0b-4a11-42b1-a79c-18e1e7d554c3',
```
Comment (AkhtarAmir) [inline plugins/aws/connect/wisdomDomainEncrypted.spec.js:6]:
```suggestion
      assistantArn: 'arn:aws:wisdom:us-east-1:000011112222:assistant/6bebcb0b-4a11-42b1-a79c-18e1e7d554c3',
```

**Review by AkhtarAmir — CHANGES_REQUESTED** [review]

File: plugins/aws/connect/wisdomDomainEncrypted.js [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+                    if (!describeKey || describeKey.err || !describeKey.data || !describeKey.data.KeyMetadata) {
+                        helpers.addResult(results, 3,
+                            `Unable to query KMS key: ${helpers.addError(describeKey)}`,
+                            region, encryptionKey);
+                        continue;
+                    }
+
+                    currentEncryptionLevel = helpers.getEncryptionLevel(describeKey.data.KeyMetadata, helpers.ENCRYPTION_LEVELS);
+                } else {
+                    helpers.addResult(results, 3,
```
Comment (AkhtarAmir) [inline plugins/aws/connect/wisdomDomainEncrypted.js:79]: Encryption is enabled by default so current encryption level is 2

**Review by AkhtarAmir — CHANGES_REQUESTED** [review]

File: plugins/aws/connect/wisdomDomainEncrypted.js (hunk 3 above)
Comment (m-nouman-suleman) [reply plugins/aws/connect/wisdomDomainEncrypted.js:79]: Can you explain a bit more?

File: plugins/aws/connect/wisdomDomainEncrypted.js [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+
+        async.each(regions.wisdom, function(region, rcb){
+            var listDomains = helpers.addSource(cache, source,
+                ['wisdom', 'listAssistants', region]);
+
+            if (!listDomains) return rcb();
+
+            if (listDomains.err || !listDomains.data) {
+                helpers.addResult(results, 3,
+                    'Unable to query wisdom domain: ' + helpers.addError(listDomains), region);
```
Comment (AkhtarAmir) [inline plugins/aws/connect/wisdomDomainEncrypted.js:46]:
```suggestion
                    'Unable to query Connect Wisdom domains: ' + helpers.addError(listDomains), region);
```

File: plugins/aws/connect/wisdomDomainEncrypted.js [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+            if (!listDomains) return rcb();
+
+            if (listDomains.err || !listDomains.data) {
+                helpers.addResult(results, 3,
+                    'Unable to query Connect Wisdom domains: ' + helpers.addError(listDomains), region);
+                return rcb();
+            }
+
+            if (!listDomains.data.length) {
+                helpers.addResult(results, 0, 'No wisdom domain found', region);
```
Comment (AkhtarAmir) [inline plugins/aws/connect/wisdomDomainEncrypted.js:51]:
```suggestion
                helpers.addResult(results, 0, 'No Connect Wisdom domains found', region);
```

**Review by AkhtarAmir — APPROVED** [review]

**Review by dipsubha06 — APPROVED** [review]

**Pull request merged**